### DISCUSSION TITLE
Search Results Linking to Incorrect Headings in Quarto

### DISCUSSION THREAD

**Discussion by dtufood-kihen** [quarto-dev/quarto-cli#10793] [Q&A]
### Description

I’m encountering an issue with the search functionality in Quarto. When I click on an item in the search results, it directs me to the section above the target rather than the specific item I clicked on.

Scenario:

* I have a document with a section that contains hundreds of subheadings.
* When I search for a specific subheading, especially one near the bottom, the search result links me to the heading directly above it, which is often far from the actual item I searched for.
* For example, if I search for a subheading near the end of the section, the anchor link takes me to a location nowhere near the subheading I'm looking for.
  
Has anyone else experienced this issue? Is there a way to ensure the search results link to the correct heading directly / item?

I’m using Quarto 1.5.57 on Ubuntu 22.04.4 LTS.

Example:

Here is one of my documents. In this case, when I search on a field name, it always links to the `### Schema for: {resource.name}` heading, rather than to a field one level down.

````qmd
---
title: "Data Dictionary"
toc: true
toc-depth: 4
number-sections: true
---

# Data Package

## Dataset

```{python}
#| echo: false
#| output: asis

from frictionless import Package
from IPython.display import Markdown

# Load the Data Package
package = Package("datapackage.yaml")

# Display the metadata using attributes
metadata = {
    "Title": package.title or "N/A",
    "Name": package.name or "N/A",
    "Description": package.description or "N/A"
}

# Display each metadata field as markdown
for key, value in metadata.items():
    display(Markdown(f"- **{key}**: {value}"))
```

## Schemas

Each resource in the dataset contains specific fields with corresponding types, descriptions, and constraints.

```{python}
#| echo: false
#| output: asis

# Loop through resources and their fields to show the schema, rendering as markdown
for resource in package.resources:
    display(Markdown(f"\n### Schema for: {resource.name}"))
    
    if not resource.schema:
        display(Markdown(f"*The **{resource.name}** resource does not have a schema definition.*"))
    else:
        for field in resource.schema.fields:
            # Display basic field information
            display(Markdown(f"\n#### Field: {field.title or field.name}"))
            display(Markdown(f"- **Description**: `{field.description or 'N/A'}`"))
            display(Markdown(f"- **Title**: {field.title or 'N/A'}"))
            display(Markdown(f"- **Name**: {field.name}"))
            display(Markdown(f"- **Type**: {field.type}"))

            # Display primary key information
            if field.name in resource.schema.primary_key:
                display(Markdown(f"- **Primary key**: Yes"))

            # Check if the field has categories
            is_categorical = 'categories' in field.custom
            display(Markdown(f"- **Categorical**: {'Yes' if is_categorical else 'No'}"))

            # Display if categories are ordered
            if is_categorical and 'categoriesOrdered' in field.custom:
                display(Markdown(f"- **Categories Ordered**: {'Yes' if field.custom['categoriesOrdered'] else 'No'}"))
```
````

**Comment by mcanouil** [comment]
Could you properly format your post using code blocks for code and terminal outputs? Thanks.
If your code contains code blocks, you need to enclose it using more backticks, *i.e.*, usually four ` ```` `.
See <https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us>.

Additionally, your example is not reproducible as there are no search engine set.
Please do ensure you share a small reproducible example, see <https://quarto.org/bug-reports.html>.

**Reply by dtufood-kihen** [reply]
> Could you properly format your post using code blocks for code and terminal outputs? Thanks. If your code contains code blocks, you need to enclose it using more backticks, _i.e._, usually four `` ```` ``. See https://quarto.org/bug-reports.html#formatting-make-githubs-markdown-work-for-us.

Fixed - my bad.

> Additionally, your example is not reproducible as there are no search engine set. Please do ensure you share a small reproducible example, see https://quarto.org/bug-reports.html.

That's pretty hard, since the document is dynamically generated based on a file, which I can't share. Also I don't know what you mean by "there are no search engine set" ?

**Reply by mcanouil** [reply]
If I user your current document, there is no search bar or anything.
Search is a website feature.

Unfortunately, if we can't reproduce, we can't help or fix a bug if there is one.

> That's pretty hard, since the document is dynamically generated based on a file, which I can't share. Also I don't know what you mean by "there are no search engine set" ?

Can't you make a fake file that mimic yours?
We don't need your files, we need the means to reproduce, and it is best if the example is small.
